### DISCUSSION TITLE
HomeAssistant history graph slowness

### DISCUSSION THREAD

**Discussion by btarb24** [emporia-vue-local/esphome#180] [General]
The esphome emporia integration with homeassistant works great.  However, if i put all 16 nodes on a single graph then I have to limit it to only show a handful of hours or else the graph gets unusably slow to react to mouse input.  Ideally, i'd love to be able to have a 24h and a weekly graph.

I write a few sensors to influxDB for long-term persistence and that remains super snappy..  do i just need to write all my sensor data there? (hoping not).  I kinda like that most sensor data is naturally purged by the HA recorder every 10 days.

Anyone have any tactics for improving the graph performance in HomeAssistant?  Or, what alternative setups do you use?

**Comment by flaviut** [comment]
Homeassistant is not well-optimized for this sort of thing. Basically just a fact of life with it's data storage architecture (sqlite vs a purpose-specific time-series database like influx).

> I write a few sensors to influxDB for long-term persistence and that remains super snappy.. do i just need to write all my sensor data there? (hoping not).

Yup. Pretty much what I do.

> I kinda like that most sensor data is naturally purged by the HA recorder every 10 days.

I believe you can configure this in influxdb: [v2+](https://docs.influxdata.com/influxdb/v2.7/reference/internals/data-retention/#bucket-retention-period), [v1](https://community.influxdata.com/t/delete-specific-data-older-than-30-days/11981).

>  Or, what alternative setups do you use?

I have a similar setup with regards to influxdb, although I use grafana as the frontend. I keep all my data permanently, how do you know I won't want to run a report on my yearly energy consumption 30 years down the road? ;)

![image](https://github.com/emporia-vue-local/esphome/assets/5213469/2110b6b2-3399-4f8d-9446-b069bf3aba1d)
![image](https://github.com/emporia-vue-local/esphome/assets/5213469/9bcf3387-03c2-481d-a8c7-da673eeb2cb8)

**Comment by btarb24** [comment]
alright, influx it is.  Thanks for the tip on the retention period config.

Curious, any idea how many GB per year you're consuming for all the emporia data?  I've only got a handful of days in mine so the sample size is too small to separate from the overhead.

looks like influx also has the ability to store an aggregate value. that may be interesting to reduce long term data storage after a certain window.

**Reply by flaviut** [reply]
> looks like influx also has the ability to store an aggregate value. that may be interesting to reduce long term data storage after a certain window.

Yes, absolutely. Past a week, I can't possibly see a use for data more granular than 1/min. I haven't gotten around to figuring out how to do this for myself though.

> Curious, any idea how many GB per year you're consuming for all the emporia data?

Don't know if I can give any useful numbers given all the extra stuff in here, but energy data + a ton of other HASS data I have loaded in is taking up 8.4GiB for 1.5 years of data.

**Reply by btarb24** [reply]
Oh that's not too bad at all.  At that rate we can just put off the aggregation task until the database slows down :)  Thanks for all the great first hand info.
